pkgs/development/lisp-modules/default.nix

```
@@ -0,0 +1,299 @@
+        ${pkg}/bin/${program} ${flags} < $buildScript
+      '';
+
+      # Copy compiled files to store
+      #
+      # Make sure to include '$' in regex to prevent skipping
+      # stuff like 'iolib.asdf.asd' for system 'iolib.asd'
+      #
+      # Same with '/': `local-time.asd` for system `cl-postgres+local-time.asd`
+      installPhase =
+        let
+          mkSystemsRegex = systems:
+            concatMapStringsSep "\\|" (replaceStrings ["." "+"] ["[.]" "[+]"]) systems;
+        in
+      ''
+        mkdir -pv $out
+        cp -r * $out
+
+        # Remove all .asd files except for those in `systems`.
+        find $out -name "*.asd" \
+        | grep -v "/\(${mkSystemsRegex systems}\)\.asd$" \
+        | xargs rm -fv || true
+      '';
+
+      dontPatchShebangs = true;
+
+      # Not sure if it's needed, but caused problems with SBCL
+      # save-lisp-and-die binaries in the past
+      dontStrip = true;
+
+    } // (args // {
+      src = if builtins.length (args.patches or []) > 0
+            then pkgs.applyPatches { inherit (args) src patches; }
+            else args.src;
+      patches = [];
+      propagatedBuildInputs = args.propagatedBuildInputs or []
+          ++ lispLibs ++ javaLibs ++ nativeLibs;
+    })));
+
+  # Build the set of lisp packages using `lisp`
+  # These packages are defined manually for one reason or another:
+  # - The library is not in quicklisp
+  # - The library that is in quicklisp is broken
+  # - Special build procedure such as cl-unicode, asdf
+  #
+  # They can use the auto-imported quicklisp packages as dependencies,
+  # but some of those don't work out of the box.
+  #
+  # E.g if a QL package depends on cl-unicode it won't build out of
+  # the box.
+  commonLispPackagesFor = spec:
+    let
+      build-asdf-system' = body: build-asdf-system (body // spec);
+    in pkgs.callPackage ./packages.nix {
+      inherit spec quicklispPackagesFor;
+      build-asdf-system = build-asdf-system';
+    };
+
+  # Build the set of packages imported from quicklisp using `lisp`
+  quicklispPackagesFor = spec:
+    let
+      build-asdf-system' = body: build-asdf-system (body // spec);
+    in pkgs.callPackage ./ql.nix {
+      build-asdf-system = build-asdf-system';
+    };
+
+  # Creates a lisp wrapper with `packages` installed
+  #
+  # `packages` is a function that takes `clpkgs` - a set of lisp
+  # packages - as argument and returns the list of packages to be
+  # installed
+  # TODO(kasper): assert each package has the same lisp and asdf?
+  lispWithPackagesInternal = clpkgs: packages:
+    let first = head (lib.attrValues clpkgs); in
+    (build-asdf-system {
+      inherit (first) pkg program flags faslExt asdf;
+      # See dontUnpack in build-asdf-system
+      src = null;
+      pname = first.pkg.pname;
+      version = "with-packages";
+      lispLibs = packages clpkgs;
+      systems = [];
+    }).overrideAttrs(o: {
+      nativeBuildInputs = [ pkgs.makeBinaryWrapper ];
```

```suggestion
      nativeBuildInputs = [ makeBinaryWrapper ];
```
we probably need this for cross friendliness